chore(lotteryNFT): upgrade to ERC1155

developerfred opened the pull request:
**ref:** #1

update solidity  and add License**

- [x] Event Ticket
- [x] Event TicketBatch
- [x] Function `multiBurn()`
- [x] Function `newBatchLotteryItem()`
- [x] Test for newBatchLotteryItem

developerfred committed:
chore(lotteryNFT): [WIP] upgrade to ERC1155
-[ ] ownerOf for ERC1155 on Lottery.sol
-[x] Event Ticker
-[x] Event TicketBatch
-[x] Function multiBurn
-[x] Funciton newBatchLotteryItem()

pancake-swap commented on test/lottery.test.js:
```diff
@@ -131,4 +134,28 @@ contract('Lottery', (accounts) => {
+        const LotteryContract = this.lotteryProxy.methods;
+        const async = [
+            await LotteryContract.newBatchLotteryItem([alice,bob,carol],[1, 2, 3],[10 ** 5, 10 ** 4, 10 ** 3]),
+            await LotteryContract.balanceOfBatch(
```
there was no balanceOfBatch and newBatchLotteryItem in your upgrade lottery contract

pancake-swap commented on contracts/LotteryNFT.sol:
```diff
@@ -1,38 +1,94 @@
 
     function newLotteryItem(address player, uint8[4] memory _lotteryNumbers, uint256 _amount, uint256 _issueIndex)
-        public onlyOwner
+        public onlyManager()
```
the manager address need set to lottery proxy, but lottery proxy need the nftcontract and lotterycontract, here how you set the manager to lotteryproxy before you haven't got the nft address?

developerfred replied:
I will go back to the current model

pancake-swap commented on contracts/LotteryNFT.sol:
```diff
@@ -1,38 +1,94 @@
+        
     }
 
+    function newBatchLotteryItem(uint256[] memory _ids, address player, uint8[4] memory _lotteryNumbers, uint256[] memory _amounts, uint256[] memory _issueIndexs)
```
where will the lottery contract call it?

developerfred replied:
I will add call.

ChefNyan commented on contracts/LotteryNFT.sol:
```diff
@@ -1,38 +1,94 @@
+        uint256 indexed _amount
+    );
+
+    event TicketBatch(
```
@pancake-swap do we need/use these events? These only add gas cost and not relevant since ERC1155 already incorporates events for minting

developerfred committed:
chore(lottery):
*Remove modifier onlyManager()
*add call newBatchLotteryItem() on function  multiBuy(uint256 _price, uint8[4][] memory _numbers) external [Lottery.sol]
*add new function getBalanceBatch(address[] players, uint256[] calldata _ids) external onlyOwner
*rewrite loop

ChefNyan commented on contracts/LotteryNFT.sol:
```diff
@@ -1,38 +1,69 @@
         return newItemId;
     }
 
+    function newBatchLotteryItem(uint256[] memory _ids, address player, uint8[4] memory _lotteryNumbers, uint256[] memory _amounts, uint256[] memory _issueIndexs)
```
`uint8[4] memory _lotteryNumbers`

--> doesn't it assign the same numbers for each new ticket minted?

`uint8[4][] memory _lotteryNumbers`

developerfred replied:
I fixed, thanks!

ChefNyan commented on contracts/LotteryNFT.sol:
```diff
@@ -1,38 +1,69 @@
+        returns (uint256[] memory)
+    {
+        uint256[] memory lotteryItemIds;
+        for (uint256 i = 0; i < _ids.length; i++) {
```
why not put them part of the same loop? instead of doing 2 loops?

developerfred replied:
I will do this.
